lightwood/mixer/sktime.py

```diff
@@ -16,25 +21,57 @@ class SkTime(BaseMixer):
     n_ts_predictions: int
     target: str
     supports_proba: bool
+    model_class = list
 
     def __init__(
             self, stop_after: float, target: str, dtype_dict: Dict[str, str],
-            n_ts_predictions: int, ts_analysis: Dict):
+            n_ts_predictions: int, ts_analysis: Dict, model_class: list = ('arima', 'AutoARIMA')):
+        """
+        This mixer is a wrapper around the popular time series library sktime. It exhibits different behavior compared
+        to other forecasting mixers, as it predicts based on indices in a forecasting horizon that is defined with
+        respect to the last seen data point at training time.
+        
+        Due to this, the mixer is forced to "fit_on_all" and so the latest point in the validation split given by the
+        mixer marks the difference between training data and where forecasts will start. This effectively means that for
+        correct forecasts, you need to specify how much time has passed since the aforementioned timestamp. By default,
+        it assumes predictions are for the very next timestamp post-training.
+        
+        If the task has groups (i.e. 'TimeseriesSettings.group_by' is not empty), the mixer will spawn one forecaster 
+        object per each different group observed at training time, plus an additional default forecaster fit with all data.
+        
+        :param stop_after: time budget in seconds.
+        :param target: column to forecast.
+        :param dtype_dict: dtypes of all columns in the data.
+        :param n_ts_predictions: length of forecasted horizon.
+        :param ts_analysis: dictionary with miscellaneous time series info, as generated by 'lightwood.data.timeseries_analyzer'.
+        :param model_class: sktime forecaster to use as underlying model(s). Should be a [namespace, method] list that points to a module inside `sktime.forecasting`.
+        """  # noqa
         super().__init__(stop_after)
         self.target = target
         dtype_dict[target] = dtype.float
-        self.model_class = AutoARIMA
         self.models = {}
         self.n_ts_predictions = n_ts_predictions
         self.ts_analysis = ts_analysis
-        self.forecasting_horizon = np.arange(1, self.n_ts_predictions)
-        self.cutoff_index = {}  # marks index at which training data stops and forecasting window starts
+        self.fh = ForecastingHorizon(np.arange(1, self.n_ts_predictions + 1), is_relative=True)
         self.grouped_by = ['__default'] if not ts_analysis['tss'].group_by else ts_analysis['tss'].group_by
         self.supports_proba = False
         self.stable = True
         self.prepared = False
 
+        # load sktime forecaster
+        self.model_path = model_class
+        sktime_module = importlib.import_module('.'.join(['sktime', 'forecasting', self.model_path[0]]))
```

Comment:
I myself am slightly confused by this... I think it might be more intuitive if either we have a finite set of options that we just hardcode as strings (ETS, ARIMA, etc) or at least make it a string ala `ets.AutoETS`, I mean, I guess it's in the docs though and this is already a kind-of very specific mixer so maybe not.

Reply:
I want to keep it dynamic, as people are already asking about forecasters that I'd rather not support by default (e.g. prophet). I will concede that maybe a string is simpler, will change the format.